tests/functional/api/test_objectstorage.py

```diff
@@ -1821,3 +1824,178 @@ def test_object_create_ext(self):
         props = self.api.object_get_properties(self.account, name, name)
         self.assertEqual(metadata['version'], props['version'])
         self.assertEqual(int(props['length']), size)
+
+
+class TestObjectStorageApiUsingCache(ObjectStorageApiTestBase):
+
+    def setUp(self):
+        super(TestObjectStorageApiUsingCache, self).setUp()
+        self.cache = dict()
+        self.api = ObjectStorageApi(self.ns, endpoint=self.uri,
+                                    cache=self.cache)
+
+        self.container = random_str(16)
+        self.path = random_str(16)
```

Comment:
No need for 16 characters. This function is really slow, please use it with parsimony.